src/test/scala/nl.knaw.dans.easy.bagstore/component/BagStoreSpec.scala

```diff
@@ -167,4 +169,16 @@ class BagStoreSpec extends TestSupportFixture
     Files.getPosixFilePermissions(bagInStore.resolve("data")) shouldBe dirPermissions
     Files.getPosixFilePermissions(bagInStore.resolve("data/sub")) shouldBe dirPermissions
   }
+
+  it should "result in a failure when the fetch.txt is invalid" in {
+    val uuid1 = UUID.fromString("00000000-0000-0000-0000-000000000001")
+    val fetchFile = File(testBagPrunedB.resolve("fetch.txt"))
+    val exceptionMessageSuffix = "Bag-id found in fetch.txt can not be found in the bag-store:"
+
+    Files.write(fetchFile.path, fetchFile.contentAsString.replaceAll("01", "10").getBytes())
+
+    bagStore.add(testBagPrunedB, Some(uuid1)) should matchPattern {
+      case Failure(CompositeException(errors)) if errors.head.getMessage contains exceptionMessageSuffix =>
```

Comment:
to satisfy codacy:
```suggestion
       case Failure(CompositeException(errors)) if errors.headOption.getOrElse(new Exception()).getMessage contains exceptionMessageSuffix =>
```
Perhaps extract method for the condition.